### PR TITLE
[master] Fix ipset new_set bug

### LINKED ISSUE #61620: [BUG] modules/ipset.py new_set Syntax error
**Description**
Running
`salt-call -l debug ipset.new_set name=whitelist set_type="hash:net"  comment=True counters=True`
Gives error:
`[ERROR   ] salt.loaded.int.module.cmdmod:1305 output: ipset v7.5: Syntax error: unknown inet family ['/usr/sbin/ipset', 'create', 'whitelist', 'hash:net', 'counters', 'family', [...], 'inet', 'comment']`

**Setup**
Default setup, no special configuration.

Please be as specific as possible and give set-up details.

- [ ] on-prem machine
- [ ] VM (Virtualbox, KVM, etc. please specify)
- [ ] VM running on a cloud service, please be explicit and add details
- [ ] container (Kubernetes, Docker, containerd, etc. please specify)
- [ ] or a combination, please be explicit
- [ ] jails if it is FreeBSD

  
**Steps to Reproduce the behavior**

```
# salt-call -l debug ipset.new_set name=whitelist set_type="hash:net"  comment=True counters=True family=ipv4
[DEBUG   ] Reading configuration from /etc/salt/minion
[DEBUG   ] Including configuration from '/etc/salt/minion.d/_schedule.conf'
[DEBUG   ] Reading configuration from /etc/salt/minion.d/_schedule.conf
[DEBUG   ] Including configuration from '/etc/salt/minion.d/odoopbx.conf'
[DEBUG   ] Reading configuration from /etc/salt/minion.d/odoopbx.conf
[DEBUG   ] Including configuration from '/etc/salt/minion_local.conf'
[DEBUG   ] Reading configuration from /etc/salt/minion_local.conf
[DEBUG   ] Using cached minion ID from /etc/salt/minion_id: baba4741-4bcd-44a9-a4af-3682d8224631
[DEBUG   ] Using importlib_metadata to load entry points
[DEBUG   ] Override  __grains__: <module 'salt.loaded.int.log_handlers.sentry_mod' from '/usr/local/lib/python3.8/dist-packages/salt/log/handlers/sentry_mod.py'>
14:16:19 [DEBUG   ] salt.utils.parsers:220 Configuration file path: /etc/salt/minion
14:16:19 [WARNING ] salt.utils.verify:590 Insecure logging configuration detected! Sensitive data may be logged.
14:16:19 [DEBUG   ] salt.loader:868 Grains refresh requested. Refreshing grains.
14:16:19 [DEBUG   ] salt.config:1917 Reading configuration from /etc/salt/minion
14:16:19 [DEBUG   ] salt.config:2080 Including configuration from '/etc/salt/minion.d/_schedule.conf'
14:16:19 [DEBUG   ] salt.config:1917 Reading configuration from /etc/salt/minion.d/_schedule.conf
14:16:19 [DEBUG   ] salt.config:2080 Including configuration from '/etc/salt/minion.d/odoopbx.conf'
14:16:19 [DEBUG   ] salt.config:1917 Reading configuration from /etc/salt/minion.d/odoopbx.conf
14:16:19 [DEBUG   ] salt.config:2080 Including configuration from '/etc/salt/minion_local.conf'
14:16:19 [DEBUG   ] salt.config:1917 Reading configuration from /etc/salt/minion_local.conf
14:16:19 [DEBUG   ] salt.loader.lazy:857 Override  __utils__: <module 'salt.loaded.int.grains.zfs' from '/usr/local/lib/python3.8/dist-packages/salt/grains/zfs.py'>
14:16:19 [DEBUG   ] salt.modules.network:2132 Elapsed time getting FQDNs: 0.009395122528076172 seconds
14:16:19 [DEBUG   ] salt.loaded.int.grains.extra:56 Loading static grains from /etc/salt/grains
14:16:20 [DEBUG   ] salt.utils.lazy:99 LazyLoaded zfs.is_supported
14:16:20 [DEBUG   ] salt.minion:798 Connecting to master. Attempt 1 of 1
14:16:20 [DEBUG   ] salt.utils.network:2268 "localhost" Not an IP address? Assuming it is a hostname.
14:16:20 [DEBUG   ] salt.minion:238 Master URI: tcp://127.0.0.1:4506
14:16:20 [DEBUG   ] salt.crypt:505 Initializing new AsyncAuth for ('/etc/salt/pki/minion', 'baba4741-4bcd-44a9-a4af-3682d8224631', 'tcp://127.0.0.1:4506')
14:16:20 [DEBUG   ] salt.transport.zeromq:394 Generated random reconnect delay between '1000ms' and '11000ms' (10505)
14:16:20 [DEBUG   ] salt.transport.zeromq:401 Setting zmq_reconnect_ivl to '10505ms'
14:16:20 [DEBUG   ] salt.transport.zeromq:405 Setting zmq_reconnect_ivl_max to '11000ms'
14:16:20 [DEBUG   ] salt.transport.zeromq:156 Connecting the Minion to the Master URI (for the return server): tcp://127.0.0.1:4506
14:16:20 [DEBUG   ] salt.transport.zeromq:1180 Trying to connect to: tcp://127.0.0.1:4506
14:16:20 [DEBUG   ] salt.crypt:219 salt.crypt.get_rsa_pub_key: Loading public key
14:16:20 [DEBUG   ] salt.crypt:942 Decrypting the current master AES key
14:16:20 [DEBUG   ] salt.crypt:211 salt.crypt.get_rsa_key: Loading private key
14:16:20 [DEBUG   ] salt.crypt:191 salt.crypt._get_key_with_evict: Loading private key
14:16:20 [DEBUG   ] salt.crypt:869 Loaded minion key: /etc/salt/pki/minion/minion.pem
14:16:20 [DEBUG   ] salt.crypt:219 salt.crypt.get_rsa_pub_key: Loading public key
14:16:20 [DEBUG   ] salt.transport.zeromq:177 Closing AsyncZeroMQReqChannel instance
14:16:20 [DEBUG   ] salt.transport.zeromq:462 Connecting the Minion to the Master publish port, using the URI: tcp://127.0.0.1:4505
14:16:20 [DEBUG   ] salt.crypt:211 salt.crypt.get_rsa_key: Loading private key
14:16:20 [DEBUG   ] salt.crypt:869 Loaded minion key: /etc/salt/pki/minion/minion.pem
14:16:20 [DEBUG   ] salt.pillar:69 Determining pillar cache
14:16:20 [DEBUG   ] salt.crypt:505 Initializing new AsyncAuth for ('/etc/salt/pki/minion', 'baba4741-4bcd-44a9-a4af-3682d8224631', 'tcp://127.0.0.1:4506')
14:16:20 [DEBUG   ] salt.transport.zeromq:156 Connecting the Minion to the Master URI (for the return server): tcp://127.0.0.1:4506
14:16:20 [DEBUG   ] salt.transport.zeromq:1180 Trying to connect to: tcp://127.0.0.1:4506
14:16:20 [DEBUG   ] salt.crypt:211 salt.crypt.get_rsa_key: Loading private key
14:16:20 [DEBUG   ] salt.crypt:869 Loaded minion key: /etc/salt/pki/minion/minion.pem
14:16:20 [DEBUG   ] salt.transport.zeromq:177 Closing AsyncZeroMQReqChannel instance
14:16:20 [DEBUG   ] salt.utils.lazy:99 LazyLoaded jinja.render
14:16:20 [DEBUG   ] salt.utils.lazy:99 LazyLoaded yaml.render
14:16:20 [DEBUG   ] salt.utils.lazy:99 LazyLoaded ipset.new_set
14:16:20 [DEBUG   ] salt.utils.lazy:99 LazyLoaded direct_call.execute
14:16:20 [DEBUG   ] salt.utils.lazy:99 LazyLoaded cmd.run
14:16:20 [INFO    ] salt.loaded.int.module.cmdmod:417 Executing command /usr/sbin/ipset in directory '/root'
14:16:20 [ERROR   ] salt.loaded.int.module.cmdmod:877 Command '/usr/sbin/ipset' failed with return code: 1
14:16:20 [ERROR   ] salt.loaded.int.module.cmdmod:879 stdout: ipset v7.5: Syntax error: unknown inet family ['/usr/sbin/ipset', 'create', 'whitelist', 'hash:net', 'counters', 'family', [...], 'inet', 'comment']
14:16:20 [ERROR   ] salt.loaded.int.module.cmdmod:883 retcode: 1
14:16:20 [ERROR   ] salt.loaded.int.module.cmdmod:1300 Command '/usr/sbin/ipset' failed with return code: 1
14:16:20 [ERROR   ] salt.loaded.int.module.cmdmod:1305 output: ipset v7.5: Syntax error: unknown inet family ['/usr/sbin/ipset', 'create', 'whitelist', 'hash:net', 'counters', 'family', [...], 'inet', 'comment']
14:16:20 [DEBUG   ] salt.crypt:505 Initializing new AsyncAuth for ('/etc/salt/pki/minion', 'baba4741-4bcd-44a9-a4af-3682d8224631', 'tcp://127.0.0.1:4506')
14:16:20 [DEBUG   ] salt.transport.zeromq:156 Connecting the Minion to the Master URI (for the return server): tcp://127.0.0.1:4506
14:16:20 [DEBUG   ] salt.transport.zeromq:1180 Trying to connect to: tcp://127.0.0.1:4506
14:16:20 [DEBUG   ] salt.transport.zeromq:177 Closing AsyncZeroMQReqChannel instance
14:16:20 [DEBUG   ] salt.utils.lazy:99 LazyLoaded nested.output
local:
    ipset v7.5: Syntax error: unknown inet family ['/usr/sbin/ipset', 'create', 'whitelist', 'hash:net', 'counters', 'family', [...], 'inet', 'comment']
```

**Versions Report**
```
# salt-call  --versions-report
Salt Version:
          Salt: 3004
 
Dependency Versions:
          cffi: Not Installed
      cherrypy: 18.6.1
      dateutil: 2.7.3
     docker-py: Not Installed
         gitdb: Not Installed
     gitpython: Not Installed
        Jinja2: 2.11.2
       libgit2: Not Installed
      M2Crypto: 0.31.0
          Mako: 1.0.7
       msgpack: 1.0.3
  msgpack-pure: Not Installed
  mysql-python: Not Installed
     pycparser: Not Installed
      pycrypto: Not Installed
  pycryptodome: 3.14.1
        pygit2: Not Installed
        Python: 3.8.10 (default, Nov 26 2021, 20:14:08)
  python-gnupg: Not Installed
        PyYAML: 5.3.1
         PyZMQ: 21.0.2
         smmap: Not Installed
       timelib: Not Installed
       Tornado: 4.5.3
           ZMQ: 4.3.3
 
System Versions:
          dist: ubuntu 20.04 focal
        locale: utf-8
       machine: x86_64
       release: 5.4.0-99-generic
        system: Linux
       version: Ubuntu 20.04 focal
 ```

</details>

### PR DESCRIPTION
**Pull request by sbourkeostk**
### What does this PR do?
Fixes a bug in ipset introduced in [0e46c149](https://github.com/saltstack/salt/commit/0e46c14907f1c63c9bd7206ed3a7dd59e59e0800#diff-f0cc809208d5a7ae3c8d9190536575ef7a8f5141d6785056ba2703991f971346L332-R332)

### What issues does this PR fix or reference?
Fixes: #61620, #62255, #63224

### Previous Behavior
Creating an ipset would fail if the set type used the `family` option.

### New Behavior
The ipset is created as expected.

### Merge requirements satisfied?
**[NOTICE] Bug fixes or features added to Salt require tests.**

- [ ] Docs
- [x] Changelog - https://docs.saltproject.io/en/master/topics/development/changelog.html
- [x] Tests written/updated

### Commits signed with GPG?
No

Please review [Salt's Contributing Guide](https://docs.saltproject.io/en/master/topics/development/contributing.html) for best practices.

See GitHub's [page on GPG signing](https://help.github.com/articles/signing-commits-using-gpg/) for more information about signing commits with GPG.

### CODE REVIEW

File: tests/pytests/functional/modules/test_ipset.py [hunk 1]
```diff
@@ -47,3 +47,19 @@ def test_ipset_add_comment_kwarg(ipset, setup_set):
...
+    """
+    test ipset.new_set with set_type that uses family (eg. hash:ip)
+    """
+    set_name = "test_name_haship"
+    ret = ipset.new_set(name=set_name, set_type="hash:ip")
+    assert ret == True
+    check_set = ipset.list_sets()
+    try:
+        assert any([x for x in check_set if x["Name"] == set_name])
+    except Exception as e:
```
Comment (Ch3LL) [inline tests/pytests/functional/modules/test_ipset.py:62]: Can you remove catching the exception here. The test will properly raise any assertion errors and call the finally.

**Review by Ch3LL — CHANGES_REQUESTED** [review]

File: tests/pytests/functional/modules/test_ipset.py (hunk 1 above)
Comment (sbourkeostk) [reply tests/pytests/functional/modules/test_ipset.py]: You are right. Done.

**Review by Ch3LL — DISMISSED** [review]

**Comment by sbourkeostk** [comment]
Incorporated suggestions from CI workflow: pre-commit checks (use f-strings) and lint

**Review by Ch3LL — APPROVED** [review]

**Comment by sbourkeostk** [comment]
Hi @Ch3LL, just wanted to check in case I'm blocking this - are you waiting on anything from me here?

**Comment by s0undt3ch** [comment]
## 💚 All backports created successfully

| Status | Branch | Result |
|:------:|:------:|:------|
|✅|3006.x|[](https://github.com/saltstack/salt/pull/65276)|

### Questions ?
Please refer to the [Backport tool documentation](https://github.com/sqren/backport)

**Pull request merged**